resolwe/process/fields.py

```diff
@@ -582,6 +637,20 @@ def entity_name(self):
         """Entity name."""
         return self._get("__entity_name")
 
+    @property
+    def relations(self):
+        """Relations."""
+        return self._get("__relations")
+
+    def get_relations(self):
+        """Get this data's relations."""
+        relations = set()
+        for relation in self._field.process.relations:
+            for partition in relation["partitions"]:
+                if partition["entity_id"] == self.entity_id:
+                    relations.add(RelationDescriptor.from_dict(relation))
+        return list(relations)
```

Comment:
How does Python compare `RelationDescriptors`? Do you have to add `__eq__` method to work as expected? I imagine that otherwise, they are compared by `id`, and even equal relations are treated as different.

Reply:
Good catch! Yes, one needs to add `__eq__` and also `__hash__` methods for objects to be usable in sets. I've added `relation.id` to the "serializer" and used it in the above methods.